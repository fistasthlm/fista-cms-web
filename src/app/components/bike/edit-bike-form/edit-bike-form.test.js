import React from 'react';
import { shallow } from 'enzyme';
import { Map } from 'immutable';
import EditBikeForm from './edit-bike-form';

const setup = () => {
    const props = Object.assign({
        bike: Map({
            title: 'dolan',
            frame: 'dolan',
            fork: 'alpina',
            cranks: 'dura ace',
            pedals: 'time',
            drivetrain: 'kmc',
            handlebars: 'easton',
            saddle: 'slr',
            frontWheel: 'halo',
            rearWheel: 'halo',
            images: [],
            instagram: 'fistasthlm',
        })
    });
    return shallow(<EditBikeForm {...props} />);
};

describe('Render', () => {
    it('should render component', () => {
        const wrapper = setup();

        expect(wrapper).toMatchSnapshot();
    });
});


describe('State', () => {
    it('should handle changes', () => {
        const wrapper = setup();

        const mockEvent = {
            target: {
                name: 'test',
                value: 'testvalue'
            }
        };

        wrapper.instance().handleChange(mockEvent);

        expect(wrapper.state().test).toEqual('testvalue');
    });
});
