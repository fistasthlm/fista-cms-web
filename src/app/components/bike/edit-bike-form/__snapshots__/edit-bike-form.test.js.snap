// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<form
  className="edit-bike"
>
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="0-title"
    name="title"
    placeholder="title"
    value="dolan"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="1-frame"
    name="frame"
    placeholder="frame"
    value="dolan"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="2-fork"
    name="fork"
    placeholder="fork"
    value="alpina"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="3-cranks"
    name="cranks"
    placeholder="cranks"
    value="dura ace"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="4-pedals"
    name="pedals"
    placeholder="pedals"
    value="time"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="5-drivetrain"
    name="drivetrain"
    placeholder="drivetrain"
    value="kmc"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="6-handlebars"
    name="handlebars"
    placeholder="handlebars"
    value="easton"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="7-saddle"
    name="saddle"
    placeholder="saddle"
    value="slr"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="8-frontWheel"
    name="frontWheel"
    placeholder="frontWheel"
    value="halo"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="9-rearWheel"
    name="rearWheel"
    placeholder="rearWheel"
    value="halo"
  />
  <BikeInput
    className="edit-bike__input"
    handleChange={[Function]}
    key="11-instagram"
    name="instagram"
    placeholder="instagram"
    value="fistasthlm"
  />
</form>
`;
